specifyweb/frontend/js_src/lib/components/AppResources/Editor.tsx

```diff
@@ -267,117 +379,6 @@ export function AppResourceEditor({
-                      isClone ? resourceData.id : undefined
-                    );
-                  }
-                : undefined
-            }
-            onSaving={(unsetUnloadProtect): false => {
-              loading(
-                (typeof directory.id === 'number'
-                  ? Promise.resolve(directory)
-                  : createResource('SpAppResourceDir', directory)
-                ).then(async (resourceDirectory) => {
-                  unsetUnloadProtect();
-
-                  if (appResource.isNew())
-                    appResource.set(
-                      'spAppResourceDir',
-                      resourceDirectory.resource_uri
-                    );
-
-                  const subType = f.maybe(
-                    toResource(serializeResource(appResource), 'SpAppResource'),
-                    getAppResourceType
-                  );
-                  // Set a mime type if it's not set yet
-                  if (typeof subType === 'string') {
-                    const type = appResourceSubTypes[subType];
-                    if (typeof type.name === 'string')
-                      appResource.set(
-                        'mimeType',
-                        type.mimeType ?? appResource.get('mimeType')
-                      );
-                  }
-
-                  await appResource.save();
-                  const resource = serializeResource(appResource);
-
-                  const data =
-                    typeof lastDataRef.current === 'function'
-                      ? lastDataRef.current()
-                      : lastDataRef.current;
-                  const appResourceData = deserializeResource({
-                    ...resourceData,
-                    data: data === undefined ? resourceData.data : data,
-                    spAppResource:
-                      toTable(appResource, 'SpAppResource')?.get(
-                        'resource_uri'
-                      ) ?? null,
-                    spViewSetObj:
-                      toTable(appResource, 'SpViewSetObj')?.get(
-                        'resource_uri'
-                      ) ?? null,
-                  });
-                  await appResourceData.save();
-                  if (appResource.specifyTable.name === 'SpAppResource')
-                    await clearUrlCache(
-                      getAppResourceUrl(appResource.get('name'))
-                    );
-                  await cleanup?.();
-
-                  setResourceData(
-                    serializeResource(
-                      appResourceData
-                    ) as SerializedResource<SpAppResourceData>
-                  );
-
-                  handleSaved(resource, {
-                    ...resourceDirectory,
-                    scope: getScope(resourceDirectory),
-                  });
-                })
-              );
-
-              return false;
-            }}
-          />
-        ) : undefined}
-      </>
-    ),
+    footer: footer,
```

Comment:
if `isFullScreen`, then you might want to return undefined instead of footer here to avoid duplicate footer (and then change types and usages accordingly)